nautobot/extras/tests/integration/test_plugins.py

```diff
@@ -122,3 +122,27 @@ def test_plugin_webhook_with_body(self):
         with open(os.path.join(tempfile.gettempdir(), "test_plugin_webhook_with_body"), "r") as f:
             self.assertEqual(json.loads(f.read()), {"message": "created"})
         os.remove(os.path.join(tempfile.gettempdir(), "test_plugin_webhook_with_body"))
+
+
+class PluginReturnUrlTestCase(SeleniumTestCase):
+    """
+    Integration tests for the CustomField and CustomFieldChoice models.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.user.is_superuser = True
+        self.user.save()
+        self.login(self.user.username, self.password)
+
+    def test_plugin_return_url(self):
+        """This test ensures that plugins return url for new objects is the list view."""
+        self.browser.visit(f'{self.live_server_url}{reverse("plugins:example_plugin:examplemodel_add")}')
+
+        form = self.browser.find_by_tag("form")
```

Comment:
🤦‍♂️ I don't like how we have to find the form on these views. How the next one finds the model form first is interesting because technically it should be the search in the navbar but 🤷

Reply:
So, this can probably be changed now. I originally tried looking for the URL link, but encountered problems with it finding the URL in the NAVBAR, so it wasn't really checking what I want it to. If you would like, I can remove this, as we should only have one Cancel button.

Reply:
This approach is better in my opinion than assuming we have only one cancel button. But this is more for me to consider better documented testing approaches as well as better generation code on the back end to simplify finding the form of interest.


It could/should be as simple as (this would require some unique generator, finder logic):
```py
from nautobot.dcim.forms import DeviceTypeForm
# ...
form = self.browser.find_like(DeviceTypeForm)
```